### PR TITLE
ICDS: Partition on aggregation level

### PR DESCRIPTION
**Pull request by sheelio**
@emord 

This partitions on aggregation level and also fixes a bug in the roll up aggregation (we were using the wrong order of columns for agg_child_health and agg_ccs_record)

### CODE REVIEW

**Comment by sheelio** [comment]
@emord 

This also removed a bunch of unused indexes (at least from the dashboard perspective).

File: custom/icds_reports/migrations/sql_templates/create_functions.sql [hunk 1]
```diff
@@ -265,17 +314,21 @@ DECLARE
...
+	_tablename3 := 'agg_child_health' || '_' || _start_date || '_3';
+	_tablename4 := 'agg_child_health' || '_' || _start_date || '_4';
+	_tablename5 := 'agg_child_health' || '_' || _start_date || '_5';
 	EXECUTE 'SELECT table_name FROM ucr_table_name_mapping WHERE table_type = ' || quote_literal('child_health_monthly') INTO _ucr_child_monthly_table;
 	_all_text = 'All';
 	_null_value = NULL;
 	_blank_value = '';
 	_no_text = 'no';
 
-	EXECUTE 'DELETE FROM ' || quote_ident(_tablename);
```
Comment (emord) [inline custom/icds_reports/migrations/sql_templates/create_functions.sql]: Why did you delete this statement? is it because this is dropped earlier?

File: custom/icds_reports/migrations/sql_templates/create_functions.sql [hunk 2]
```diff
@@ -337,23 +390,11 @@ BEGIN
...
-	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename || '_indx10') || ' ON ' || quote_ident(_tablename) || '(supervisor_id)'; -- for third query
-	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename || '_indx11') || ' ON ' || quote_ident(_tablename) || '(block_id)'; -- for fourth query
-	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename || '_indx12') || ' ON ' || quote_ident(_tablename) || '(district_id)';
-	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename || '_indx13') || ' ON ' || quote_ident(_tablename) || '(state_id)';
-	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename || '_indx14') || ' ON ' || quote_ident(_tablename) || '(aggregation_level)';
-
-
-    -- may want a double index on month and caste for aggregate location query
+	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename5 || '_indx1') || ' ON ' || quote_ident(_tablename5) || '(state_id, district_id, block_id, supervisor_id, awc_id)';
+	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename5 || '_indx2') || ' ON ' || quote_ident(_tablename5) || '(state_id, district_id, block_id, supervisor_id)';
```
Comment (emord) [inline custom/icds_reports/migrations/sql_templates/create_functions.sql]: You should delete indx2. Because it starts with the same IDs, all queries that need index2 should be able to use indx1

File: custom/icds_reports/migrations/sql_templates/create_functions.sql [hunk 3]
```diff
@@ -411,16 +453,21 @@ BEGIN
...
-		'FROM ' || quote_ident(_tablename) || ' ' ||
-		'GROUP BY state_id, district_id, block_id, supervisor_id, month, gender, age_tranche, caste, disabled, minority, resident)';
-
-	EXECUTE 'INSERT INTO ' || quote_ident(_tablename) || '(SELECT ' ||
+		'4, ' ||
+		_rollup_text2 ||
+		'FROM ' || quote_ident(_tablename5) || ' ' ||
+		'GROUP BY state_id, district_id, block_id, supervisor_id, month, gender, age_tranche)';
+	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename4 || '_indx1') || ' ON ' || quote_ident(_tablename4) || '(state_id, district_id, block_id, supervisor_id)';
+	EXECUTE 'CREATE INDEX ' || quote_ident(_tablename4 || '_indx2') || ' ON ' || quote_ident(_tablename4) || '(state_id, district_id, block_id)';
```
Comment (emord) [inline custom/icds_reports/migrations/sql_templates/create_functions.sql]: Can delete indx2 here

**Review by emord — COMMENTED** [review]
Thanks for doing this! I think you can delete any indx2 in the function script. otherwise looks good

File: custom/icds_reports/migrations/sql_templates/create_functions.sql (hunk 1 above)
Comment (sheelio) [inline custom/icds_reports/migrations/sql_templates/create_functions.sql]: Yup, these tables are already dropped so didn't think it was needed;

**Comment by sheelio** [comment]
Removed the index 2s so this should be good to merge

**Review by emord — APPROVED** [review]
